maintainer/channels.py

```diff
@@ -0,0 +1,126 @@
+        channels_detailed_info = {}
+        for channel in channels_basic_info:
+            channels_detailed_info[channel["cid"]] = self._requests_get(
+                'channels/{}'.format(channel["cid"]))
+        return channels_detailed_info
+
+    def channels_to_quarantine(self):
+        '''
+        Method is using channels_detailed_information for getting seconds_empty
+        and channel_topic values in order to specify if there should be put
+        status of quarantine on channels that meet certain condition.
+        '''
+        channels_detailed_info = self.channels_detailed_information()
+        for channel_id in channels_detailed_info:
+            channel = channels_detailed_info[channel_id]
+            if int(channel["seconds_empty"]) > self.qtime and 'protected' not in channel[
+                    "channel_topic"]:
+                payload = {'channel_topic': 'quarantine'}
+                self._requests_post('channels/{}/topic'.format(channel_id), payload)
+                logging.debug("quarantined channel: {}".format(channel_id))
+
+    def delete_parents(self):
+        '''
+        Method is used to delete channels that are parents to other channels.
+        We are using this method after delete_children because it deletes
+        children of parent as well.
+        '''
+        channels_data = self.channels_detailed_information()
+        for channel_id in channels_data:
+            channel = channels_data[channel_id]
+            if 'protected' not in channel["channel_topic"] and int(channel["seconds_empty"]) > self.qtime and 'quarantine' in channel["channel_topic"]:
+                self._requests_delete('channels/{}'.format(channel_id))
+            logging.debug("deleted channel {}".format(channel_id))
+
+    def delete_children(self):
+        '''
+        Method is used for deleting channels that are having children status.
+        '''
+        channels_data = self.channels_detailed_information()
+        children_list = self.list_of_children()
+        for cid in channels_data:
+            if cid in children_list:
+                if 'protected' not in channels_data[cid]["channel_topic"] and int(channels_data[cid]["seconds_empty"]) > self.qtime and 'quarantine' in channels_data[cid]["channel_topic"]:
+                    requests.delete('{}/channels/{}'.format(
+                        self.url, cid), auth=(self.username, self.password))
+                    logging.debug("deleted children channel {}".format(cid))
+
+    def list_of_children(self):
+        '''
+        Method is used for getting list of children channels on server.
+        '''
+        channels_data = self.channels_detailed_information()
+        child_list = []
+        for cid in channels_data:
+            for channel_id in channels_data:
+                if channels_data[channel_id]["pid"] == cid:
+                    child_list.append(channel_id)
+        return child_list
+
+def main():
+    channels = ChannelsMaintanance()
```

Comment:
So by default right now qtime = 2? Is it production ready?

Reply:
Right now we set both quarantine and removal of channels on 2 days, but i think that 7 days for quarantine and 7 days for removal will be good idea.